### DISCUSSION TITLE
Accessing one element of an array

### DISCUSSION THREAD

**Discussion by Oassal** [FreeOpcUa/python-opcua#1517] [Q&A]
Hello! 

I am trying to write on specific addresses in an array (i have a node that has an array of ~16,000 integers and i am only interested in some of them), after doing some research, i found that a possible solution is to pass indexRange in the writing service, however, i could not find anything related to that in python-opcua documentation/examples

Is that parameter available in python-opcua? If it is not, do you have any ideas? 

Thanks alot 😄

**Comment by schroeder-** [comment]
If you switch to asyncua you can use  
`async def read_attribute(self, attr, indexrange=None, raise_on_bad_status=True):`

**Reply by Oassal** [reply]
Thanks for the suggestion, i succsefully read/wrote the values of the whole array but i couldn't access a specific one

I used the following syntax : 
        `await var.write_attribute(ua.AttributeIds.Value,new_value,indexrange="1")`

to change the second element of the following array: 
![image](https://user-images.githubusercontent.com/125380070/233049246-30b27cd0-975e-47cd-83c0-814e20087a67.png)

The result : it replaces the whole array with a single integer: 
![image](https://user-images.githubusercontent.com/125380070/233049518-8080356e-59b6-466d-a30c-58347f000ddc.png)

Any ideas ?

**Reply by schroeder-** [reply]
I didn't use index range, so i can help here. Some servers may not handle index range and ignore it.

**Reply by eduardo-leite-oliveira** [reply]
Hi Oassal,

Could you help me on how did you read the values of the whole array?

**Reply by Oassal** [reply]
Hello @eduardo-leite-oliveira,

You can simply read the whole array by using the same syntax used for reading simple variables : 

```python
        client=Client(url)
        client.connect()
        Array = client.get_node("ns= 1; i=1") #the array is represented by the namespace ns= 1 and has the identifier 1
```
I hope this helped
